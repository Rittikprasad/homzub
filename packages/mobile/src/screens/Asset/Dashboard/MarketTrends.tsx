import React from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { CommonParamList } from '@homzhub/mobile/src/navigation/Common';
import AssetMarketTrends from '@homzhub/mobile/src/components/molecules/AssetMarketTrends';
import { UserScreen } from '@homzhub/mobile/src/components/HOC/UserScreen';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

type libraryProps = NavigationScreenProps<CommonParamList, ScreensKeys.MarketTrends>;
type Props = WithTranslation & libraryProps;

export class MarketTrends extends React.PureComponent<Props> {
  public render = (): React.ReactElement => {
    const { t, navigation, route } = this.props;
    const title = route.params && route.params.isFromDashboard ? t('assetDashboard:dashboard') : t('assetMore:more');

    return (
      <UserScreen
        title={title}
        pageTitle={t('assetMore:marketTrends')}
        onBackPress={navigation.goBack}
        scrollEnabled={false}
      >
        <AssetMarketTrends onTrendPress={this.openWebView} />
      </UserScreen>
    );
  };

  private openWebView = (url: string, trendId: number): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.WebViewScreen, { url, trendId });
  };
}

export default withTranslation(LocaleConstants.namespacesKey.assetDashboard)(MarketTrends);
