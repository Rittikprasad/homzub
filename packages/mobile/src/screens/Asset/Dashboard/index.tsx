import React from 'react';
import { StyleProp, ViewStyle } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { SvgProps } from 'react-native-svg';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { FunctionUtils } from '@homzhub/common/src/utils/FunctionUtils';
import { AnalyticsService } from '@homzhub/common/src/services/Analytics/AnalyticsService';
import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { EventType } from '@homzhub/common/src/services/Analytics/EventType';
import { AssetRepository } from '@homzhub/common/src/domain/repositories/AssetRepository';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { AssetActions } from '@homzhub/common/src/modules/asset/actions';
import { PortfolioActions } from '@homzhub/common/src/modules/portfolio/actions';
import { RecordAssetActions } from '@homzhub/common/src/modules/recordAsset/actions';
import { UserActions } from '@homzhub/common/src/modules/user/actions';
import { SearchActions } from '@homzhub/common/src/modules/search/actions';
import { CommonSelectors } from '@homzhub/common/src/modules/common/selectors';
import { SearchSelector } from '@homzhub/common/src/modules/search/selectors';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import AddProperty from '@homzhub/common/src/assets/images/addProperty.svg';
import Alert from '@homzhub/common/src/assets/images/alert.svg';
import CustomSearch from '@homzhub/common/src/assets/images/customSearch.svg';
import Finance from '@homzhub/common/src/assets/images/finance.svg';
import Help from '@homzhub/common/src/assets/images/headphone.svg';
import ServiceRequest from '@homzhub/common/src/assets/images/serviceRequest.svg';
import {
  AssetAdvertisementBanner,
  AssetMetricsList,
  AssetSummary,
  FullScreenAssetDetailsCarousel,
} from '@homzhub/mobile/src/components';
import AssetMarketTrends from '@homzhub/mobile/src/components/molecules/AssetMarketTrends';
import IconSheet, { ISheetData } from '@homzhub/mobile/src/components/molecules/IconSheet';
import UserSubscriptionPlan from '@homzhub/common/src/components/molecules/UserSubscriptionPlan';
import WelcomeCard from '@homzhub/common/src/components/molecules/WelcomeCard';
import FinanceOverview from '@homzhub/mobile/src/components/organisms/FinanceOverview';
import LeasePropertyList from '@homzhub/mobile/src/components/organisms/LeasePropertyList';
import PendingPropertyListCard from '@homzhub/mobile/src/components/organisms/PendingPropertyListCard';
import VisitListCard from '@homzhub/mobile/src/components/organisms/VisitListCard';
import { UserScreen } from '@homzhub/mobile/src/components/HOC/UserScreen';
import { Asset, PropertyStatus } from '@homzhub/common/src/domain/models/Asset';
import { Filters } from '@homzhub/common/src/domain/models/AssetFilter';
import { IActions, ISelectedAssetPlan } from '@homzhub/common/src/domain/models/AssetPlan';
import { IMetricsData } from '@homzhub/common/src/domain/models/AssetMetrics';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { DashboardNavigatorParamList } from '@homzhub/mobile/src/navigation/DashboardStack';
import { Attachment } from '@homzhub/common/src/domain/models/Attachment';
import { Currency } from '@homzhub/common/src/domain/models/Currency';
import { IFilter } from '@homzhub/common/src/domain/models/Search';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { ISetAssetPayload } from '@homzhub/common/src/modules/portfolio/interfaces';
import { IAssetVisitPayload, VisitStatus } from '@homzhub/common/src/domain/repositories/interfaces';
import { IApiClientError } from '@homzhub/common/src/network/ApiClientError';

interface IDispatchProps {
  setCurrentFilter: (payload: Filters) => void;
  setAssetId: (payload: number) => void;
  setSelectedPlan: (payload: ISelectedAssetPlan) => void;
  setAddPropertyFlow: (payload: boolean) => void;
  setCurrentAsset: (payload: ISetAssetPayload) => void;
  getAssetVisit: (payload: IAssetVisitPayload) => void;
  setInitialState: () => void;
  getTenanciesDetails: () => void;
  getFilterDetails: (payload: IFilter) => void;
  setFilter: (payload: IFilter) => void;
  setEditPropertyFlow: (payload: boolean) => void;
  toggleEditPropertyFlowBottomSheet: (payload: boolean) => void;
}

interface IReduxStateProps {
  assets: Asset[];
  filters: IFilter;
  defaultCurrency: Currency;
}

type libraryProps = NavigationScreenProps<DashboardNavigatorParamList, ScreensKeys.DashboardLandingScreen>;
type Props = WithTranslation & libraryProps & IDispatchProps & IReduxStateProps;

interface IDashboardState {
  metrics: IMetricsData;
  pendingProperties: Asset[];
  isLoading: boolean;
  showBottomSheet: boolean;
}

enum MetricType {
  OFFER = 'OFFER',
  SITE_VISITS = 'SITE_VISIT',
  SHORTLISTED = 'SHORTLISTED',
}

const ShowInMvpRelease = false;

export class Dashboard extends React.PureComponent<Props, IDashboardState> {
  public focusListener: any;
  public blurListener: any;

  public state = {
    metrics: {} as IMetricsData,
    pendingProperties: [],
    isLoading: false,
    showBottomSheet: false,
  };

  public componentDidMount = (): void => {
    const { navigation } = this.props;
    this.focusListener = navigation.addListener('focus', () => {
      this.getScreenData().then();
    });
    this.blurListener = navigation.addListener('blur', () => {
      this.handleScreenBlur();
    });
  };

  public componentWillUnmount(): void {
    this.focusListener();
    this.blurListener();
  }

  public render = (): React.ReactElement => {
    const { t, route } = this.props;
    const { isLoading, pendingProperties, metrics } = this.state;

    return (
      <>
        <UserScreen loading={isLoading} isGradient title={t('dashboard')} onPlusIconClicked={this.openBottomSheet}>
          {this.renderAssetMetricsAndUpdates()}
          {pendingProperties.length > 0 && (
            <PendingPropertyListCard
              data={pendingProperties}
              onPressComplete={this.onCompleteDetails}
              onSelectAction={this.handleActionSelection}
              onViewProperty={this.onViewProperty}
            />
          )}
          {!metrics?.metricValues?.assets?.count && <WelcomeCard handlePress={this.handleAddProperty} />}
          {metrics?.isTenant && <LeasePropertyList />}
          {metrics?.isTenant && <VisitListCard />}
          {!metrics?.isTenant && <FinanceOverview />}
          <AssetMarketTrends isDashboard onViewAll={this.onViewAll} onTrendPress={this.onTrendPress} />
          <AssetAdvertisementBanner />
          {ShowInMvpRelease && <UserSubscriptionPlan onApiFailure={this.onAssetSubscriptionApiFailure} />}
          {this.renderBottomSheet()}
        </UserScreen>
        {Boolean(route?.params?.imageLink && route?.params?.imageLink?.length > 0) && this.renderFullImageView()}
      </>
    );
  };

  public renderAssetMetricsAndUpdates = (): React.ReactElement => {
    const { t } = this.props;
    const { metrics } = this.state;
    return (
      <>
        <AssetMetricsList
          title={`${metrics?.metricValues?.assets?.count ?? 0}`}
          data={metrics?.metricValues?.miscellaneous ?? []}
          subscription={metrics?.metricValues?.userServicePlan?.label}
          onMetricsClicked={this.handleMetricsNavigation}
          subTitleText={metrics?.isTenant ? t('assetPortfolio:tenancies') : t('common:properties')}
        />
        <AssetSummary
          notification={metrics?.updates?.notifications?.count ?? 0}
          serviceTickets={metrics?.updates?.tickets?.open?.count ?? 0}
          dues={metrics?.updates?.dues?.count ?? 0}
          messages={metrics?.updates?.messages?.unread?.count ?? 0}
          containerStyle={styles.assetCards()}
          onPressServiceTickets={this.handleServiceTickets}
          onPressNotification={this.handleNotification}
          onPressMessages={this.handleMessages}
        />
      </>
    );
  };

  public renderBottomSheet = (): React.ReactElement => {
    const { showBottomSheet } = this.state;
    const data = this.formatBottomSheetData();

    return <IconSheet data={data} onCloseSheet={this.closeBottomSheet} isVisible={showBottomSheet} numOfColumns={3} />;
  };

  public renderFullImageView = (): React.ReactElement | null => {
    const { route } = this.props;

    if (route?.params?.imageLink) {
      const imageLink = ObjectMapper.deserialize(Attachment, { link: route?.params?.imageLink });

      const onPressCross = (): void => {
        this.handleScreenBlur();
      };

      return (
        <>
          <FullScreenAssetDetailsCarousel
            data={[imageLink]}
            activeSlide={0}
            onFullScreenToggle={onPressCross}
            hasOnlyImages
          />
        </>
      );
    }

    return null;
  };

  // HANDLERS
  private onAssetSubscriptionApiFailure = (err: IApiClientError): void => {
    AlertHelper.error({ message: ErrorUtils.getErrorMessage(err) });
  };

  private onCompleteDetails = (assetId: number): void => {
    const { navigation, setAssetId, setEditPropertyFlow, toggleEditPropertyFlowBottomSheet } = this.props;
    setAssetId(assetId);
    setEditPropertyFlow(true);
    toggleEditPropertyFlowBottomSheet(true);
    // @ts-ignore
    navigation.navigate(ScreensKeys.PropertyPostStack, {
      screen: ScreensKeys.AddProperty,
      params: { previousScreen: ScreensKeys.Dashboard },
    });
  };

  private onViewAll = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.MarketTrends, { isFromDashboard: true });
  };

  private onTrendPress = (url: string, trendId: number): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.WebViewScreen, { url, trendId });
  };

  private onViewProperty = (data: ISetAssetPayload): void => {
    const { setCurrentAsset, navigation } = this.props;
    setCurrentAsset(data);
    navigation.navigate(ScreensKeys.PropertyDetailScreen);
  };

  private handleScreenBlur = (): void => {
    const { navigation } = this.props;
    navigation.setParams({ imageLink: undefined });
  };

  private handleMessages = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.Messages, { isFromDashboard: true });
  };

  private handleServiceTickets = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.ServiceTicketScreen, { isFromDashboard: true });
  };

  private handleAddProperty = (): void => {
    const { navigation } = this.props;
    // @ts-ignore
    navigation.navigate(ScreensKeys.PropertyPostStack, { screen: ScreensKeys.AssetLocationSearch });
    AnalyticsService.track(EventType.AddPropertyInitiation);
  };

  private handleNotification = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.AssetNotifications, { isFromDashboard: true });
  };

  private handleMetricsNavigation = (name: string): void => {
    const { navigation, setCurrentFilter, t } = this.props;

    const handleNavigation = (): void => {
      const callNavigation = (screen: ScreensKeys): void =>
        // @ts-ignore
        navigation.navigate(screen, { screenTitle: t('assetDashboard:dashboard') });

      switch (name) {
        case MetricType.OFFER:
          callNavigation(ScreensKeys.PropertyOfferList);
          break;
        case MetricType.SHORTLISTED:
          callNavigation(ScreensKeys.SavedPropertiesScreen);
          break;
        case MetricType.SITE_VISITS:
          callNavigation(ScreensKeys.PropertyVisits);
          break;
        default:
          FunctionUtils.noop();
      }
    };

    if ([MetricType.OFFER, MetricType.SHORTLISTED, MetricType.SITE_VISITS].includes(name as MetricType)) {
      handleNavigation();
      return;
    }
    setCurrentFilter(name as Filters);
    // @ts-ignore
    navigation.navigate(ScreensKeys.Portfolio, {
      screen: ScreensKeys.PortfolioLandingScreen,
      params: {
        isFromNavigation: true,
      },
    });
  };

  private handleActionSelection = (item: IActions, assetId: number): void => {
    const { navigation, setSelectedPlan, setAssetId } = this.props;
    setSelectedPlan({ id: item.id, selectedPlan: item.type });
    setAssetId(assetId);
    // @ts-ignore
    navigation.navigate(ScreensKeys.PropertyPostStack, {
      screen: ScreensKeys.AssetListing,
      params: { previousScreen: ScreensKeys.Dashboard },
    });
  };

  private openBottomSheet = (): void => this.setState({ showBottomSheet: true });

  private closeBottomSheet = (): void => {
    const { setInitialState } = this.props;
    setInitialState();
    this.setState({ showBottomSheet: false });
  };

  private formatBottomSheetData = (): ISheetData[] => {
    const { t, navigation, assets, getFilterDetails, filters, setFilter, defaultCurrency } = this.props;
    const iconSize = 38;

    const handleAddFinancialRecord = (): void => {
      if (assets.length <= 0) {
        AlertHelper.error({ message: t('assetFinancial:addProperty') });
        return;
      }
      navigation.navigate(ScreensKeys.AddRecordScreen, { isFromDashboard: true });
    };

    const handleAddServiceTicket = (): void => {
      if (assets.length <= 0) {
        AlertHelper.error({ message: t('assetDashboard:addPropertyForTicket') });
        return;
      }
      navigation.navigate(ScreensKeys.AddServiceTicket, { isFromDashboard: true });
    };

    const handleAddSupportTicket = (isBugReport?: boolean): void => {
      navigation.navigate(ScreensKeys.SupportScreen, {
        isFromDashboard: true,
        ...(isBugReport && { categoryId: 41 }), // For Bug Report hard-coding the bug category id
      });
    };

    const handleCustomSearch = (): void => {
      getFilterDetails({ asset_group: filters.asset_group });
      setFilter({ currency_code: defaultCurrency.currencyCode });
      navigation.navigate(ScreensKeys.SearchRequirement, { isFromAuth: false });
    };

    const ImageHOC = (Image: React.FC<SvgProps>): React.ReactElement => <Image width={iconSize} height={iconSize} />;

    return [
      {
        icon: ImageHOC(AddProperty),
        label: t('property:addProperty'),
        onPress: this.handleAddProperty,
      },
      {
        icon: ImageHOC(Finance),
        label: t('assetFinancial:addFinancials'),
        onPress: handleAddFinancialRecord,
      },
      {
        icon: ImageHOC(ServiceRequest),
        label: t('assetDashboard:serviceTicket'),
        onPress: handleAddServiceTicket,
      },
      {
        icon: ImageHOC(CustomSearch),
        label: t('propertySearch:shareRequirement'),
        onPress: handleCustomSearch,
      },
      {
        icon: ImageHOC(Help),
        label: t('common:askHelp'),
        onPress: handleAddSupportTicket,
      },
      {
        icon: ImageHOC(Alert),
        label: t('common:bugReport'),
        onPress: (): void => handleAddSupportTicket(true),
      },
    ];
  };

  // HANDLERS end

  // APIs
  private getScreenData = async (): Promise<void> => {
    const { getTenanciesDetails } = this.props;
    await this.getAssetMetrics();
    await this.getPendingProperties();
    this.getVisitData();
    getTenanciesDetails();
  };

  private getAssetMetrics = async (): Promise<void> => {
    this.setState({ isLoading: true });
    try {
      const { metricsData } = await DashboardRepository.getAssetMetrics('v4');
      this.setState({ metrics: metricsData, isLoading: false });
    } catch (e) {
      this.setState({ isLoading: false });
      const error = ErrorUtils.getErrorMessage(e.details);
      AlertHelper.error({ message: error });
    }
  };

  private getPendingProperties = async (): Promise<void> => {
    try {
      const response: Asset[] = await AssetRepository.getPropertiesByStatus(PropertyStatus.PENDING);
      this.setState({ pendingProperties: response });
    } catch (e) {
      AlertHelper.error({ message: ErrorUtils.getErrorMessage(e.details) });
    }
  };

  private getVisitData = (): void => {
    const { getAssetVisit } = this.props;
    getAssetVisit({
      start_date__gte: new Date().toISOString(),
      status__in: `${VisitStatus.APPROVED},${VisitStatus.PENDING}`,
    });
  };
  // APIs end
}

export const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { setCurrentFilter, setCurrentAsset, setInitialState, getTenanciesDetails } = PortfolioActions;
  const { setAddPropertyFlow } = UserActions;
  const { setAssetId, setSelectedPlan, setEditPropertyFlow, toggleEditPropertyFlowBottomSheet } = RecordAssetActions;
  const { getAssetVisit } = AssetActions;
  const { getFilterDetails, setFilter } = SearchActions;
  return bindActionCreators(
    {
      setCurrentFilter,
      setAssetId,
      setSelectedPlan,
      setAddPropertyFlow,
      setCurrentAsset,
      setInitialState,
      getAssetVisit,
      getTenanciesDetails,
      getFilterDetails,
      setFilter,
      setEditPropertyFlow,
      toggleEditPropertyFlowBottomSheet,
    },
    dispatch
  );
};

const mapStateToProps = (state: IState): IReduxStateProps => {
  const { getUserAssets } = UserSelector;
  const { getFilters } = SearchSelector;
  const { getDefaultCurrency } = CommonSelectors;
  return {
    assets: getUserAssets(state),
    filters: getFilters(state),
    defaultCurrency: getDefaultCurrency(state),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation(LocaleConstants.namespacesKey.assetDashboard)(Dashboard));

const styles = {
  assetCards: (): StyleProp<ViewStyle> => ({
    marginVertical: 12,
  }),
  evenItem: (): StyleProp<ViewStyle> => ({
    marginEnd: 17,
  }),
};
