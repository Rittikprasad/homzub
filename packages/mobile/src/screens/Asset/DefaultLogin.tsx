import React from 'react';
import { View, StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import Logo from '@homzhub/common/src/assets/images/homzhubLogo.svg';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { BottomTabNavigatorParamList } from '@homzhub/mobile/src/navigation/BottomTabs';
import { Button } from '@homzhub/common/src/components/atoms/Button';
import { Text } from '@homzhub/common/src/components/atoms/Text';

type libraryProps = WithTranslation & NavigationScreenProps<BottomTabNavigatorParamList, ScreensKeys.DefaultLogin>;
type IProps = libraryProps;

export class DefaultLogin extends React.PureComponent<IProps> {
  public render(): React.ReactElement {
    const { t } = this.props;
    return (
      <View style={styles.container}>
        <Logo height={100} width={100} />
        <Text type="small" textType="regular" style={styles.loginText}>
          {t('pleaseSignup')}
        </Text>
        <Button type="primary" title={t('signUp')} containerStyle={styles.button} onPress={this.navigateToSignup} />
      </View>
    );
  }

  public navigateToSignup = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.AuthStack, {
      screen: ScreensKeys.SignUp,
      params: { onCallback: this.navigateToDashboard },
    });
  };

  private navigateToDashboard = (): void => {
    const { navigation } = this.props;
    // @ts-ignore
    navigation.navigate(ScreensKeys.BottomTabs);
  };
}

export default withTranslation()(DefaultLogin);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.colors.white,
  },
  loginText: {
    marginVertical: 10,
  },
  button: {
    flex: 0,
    marginVertical: 10,
  },
});
