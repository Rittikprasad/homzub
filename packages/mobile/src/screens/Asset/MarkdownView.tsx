import React, { PureComponent } from 'react';
import { View, StyleSheet, ScrollView } from 'react-native';
import { WithTranslation } from 'react-i18next';
// @ts-ignore
import Markdown from 'react-native-easy-markdown';
import { PropertyPostStackParamList } from '@homzhub/mobile/src/navigation/PropertyPostStack';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { CommonService } from '@homzhub/common/src/services/CommonService';
import { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Header } from '@homzhub/mobile/src/components';

type OwnProps = WithTranslation & NavigationScreenProps<PropertyPostStackParamList, ScreensKeys.MarkdownScreen>;
type Props = OwnProps;

interface IMarkdownState {
  markdownData: string;
}

export class MarkdownView extends PureComponent<Props, IMarkdownState> {
  public state = {
    markdownData: '',
  };

  public componentDidMount = (): void => {
    const {
      route: { params },
    } = this.props;
    CommonService.getMarkdownData(params.isFrom).then((response) => {
      this.setState({ markdownData: response });
    });
  };

  public render(): React.ReactElement {
    const {
      route: { params },
    } = this.props;
    const { markdownData } = this.state;
    return (
      <View style={styles.container}>
        <Header
          type="primary"
          icon={icons.leftArrow}
          onIconPress={this.navigateBack}
          title={params.title ?? ''}
          testID="header"
        />
        <ScrollView style={styles.scrollView}>
          <View style={styles.markdownContainer}>
            <Markdown markdownStyles={{ strong: { fontWeight: 'bold' }, text: { fontWeight: 'normal', fontSize: 16 } }}>
              {markdownData}
            </Markdown>
          </View>
        </ScrollView>
      </View>
    );
  }

  public navigateBack = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  scrollView: {
    flex: 1,
    backgroundColor: theme.colors.white,
  },
  markdownContainer: {
    margin: theme.layout.screenPadding,
  },
});
