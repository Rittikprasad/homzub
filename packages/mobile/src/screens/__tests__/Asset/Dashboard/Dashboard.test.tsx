import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PortfolioActionTypes } from '@homzhub/common/src/modules/portfolio/actions';
import { Dashboard, mapDispatchToProps } from '@homzhub/mobile/src/screens/Asset/Dashboard';
import { Filters } from '@homzhub/common/src/domain/models/AssetFilter';

describe('Dashboard Screen', () => {
  let component: any;
  let props: any;

  beforeEach(() => {
    props = {
      navigation: {
        goBack: jest.fn(),
        addListener: jest.fn(),
        navigate: jest.fn(),
        dispatch: jest.fn(),
      },
      setCurrentFilter: jest.fn(),
    };
    component = shallow(<Dashboard {...props} t={(key: string): string => key} />);
  });

  it('should render snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should navigate to view all', () => {
    component.instance().onViewAll();
    expect(props.navigation.navigate).toBeCalled();
  });

  it('should navigate to notifications', () => {
    component.instance().handleNotification();
    expect(props.navigation.navigate).toBeCalled();
  });

  it('should handle metrics navigation', () => {
    component.instance().handleMetricsNavigation();
    expect(props.navigation.navigate).toBeCalled();
  });

  it('should handle mapDispatchToProps', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).setCurrentFilter(Filters.ALL);
    expect(dispatch.mock.calls[0][0]).toStrictEqual({
      type: PortfolioActionTypes.SET.CURRENT_FILTER,
      payload: Filters.ALL,
    });
  });
});
