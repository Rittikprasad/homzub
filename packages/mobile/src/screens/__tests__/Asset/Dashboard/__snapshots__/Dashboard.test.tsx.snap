// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Screen should render snapshot 1`] = `
<Fragment>
  <Memo(UserScreen)
    isGradient={true}
    loading={false}
    onPlusIconClicked={[Function]}
    title="dashboard"
  >
    <Memo(AssetMetricsList)
      data={Array []}
      onMetricsClicked={[Function]}
      subTitleText="common:properties"
      title="0"
    />
    <AssetSummary
      containerStyle={
        Object {
          "marginVertical": 12,
        }
      }
      dues={0}
      messages={0}
      notification={0}
      onPressMessages={[Function]}
      onPressNotification={[Function]}
      onPressServiceTickets={[Function]}
      serviceTickets={0}
    />
    <WelcomeCard
      handlePress={[Function]}
    />
    <Connect(withI18nextTranslation(FinanceOverview)) />
    <withI18nextTranslation(AssetMarketTrends)
      isDashboard={true}
      onTrendPress={[Function]}
      onViewAll={[Function]}
    />
    <AssetAdvertisementBanner />
    <IconSheet
      data={
        Array [
          Object {
            "icon": <UNDEFINED
              height={38}
              width={38}
            />,
            "label": "property:addProperty",
            "onPress": [Function],
          },
          Object {
            "icon": <UNDEFINED
              height={38}
              width={38}
            />,
            "label": "assetFinancial:addFinancials",
            "onPress": [Function],
          },
          Object {
            "icon": <UNDEFINED
              height={38}
              width={38}
            />,
            "label": "assetDashboard:serviceTicket",
            "onPress": [Function],
          },
          Object {
            "icon": <UNDEFINED
              height={38}
              width={38}
            />,
            "label": "propertySearch:shareRequirement",
            "onPress": [Function],
          },
          Object {
            "icon": <UNDEFINED
              height={38}
              width={38}
            />,
            "label": "common:askHelp",
            "onPress": [Function],
          },
          Object {
            "icon": <UNDEFINED
              height={38}
              width={38}
            />,
            "label": "common:bugReport",
            "onPress": [Function],
          },
        ]
      }
      isVisible={false}
      numOfColumns={3}
      onCloseSheet={[Function]}
    />
  </Memo(UserScreen)>
</Fragment>
`;
